### PR TITLE
fix:  Check service account name and type during default SA token grabbing

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Fix grabbing of default service account token in case SA has multiple assigned secrets

### What issues does this PR fix or reference?
N/A

### CODE REVIEW

**Comment by nickboldt** [comment]
From Rick:

> I tried it in the lab (transcribing from TS to JS), it works.  So I passed it to the customer, we are now past the crwctl auth problems that have been plaguing us.

> Please consider this an 'upvote' from Support for this change.  Thanks!

**Comment by tolusha** [comment]
@sleshchenko 
To fix the build pls apply the changes bellow
```
diff --git a/src/api/kube.ts b/src/api/kube.ts
index cde078e..19019d7 100644
--- a/src/api/kube.ts
+++ b/src/api/kube.ts
@@ -1061,8 +1061,8 @@ export class KubeHelper {
       throw new Error(`Unable to get default service account token since there is no secret in '${namespaceName}' namespace`)
     }
 
-    let v1DefaultSATokenSecret = v1SecretList.items.find(secret => secret.metadata!.annotations 
-      && secret.metadata!.annotations['kubernetes.io/service-account.name'] === saName 
+    let v1DefaultSATokenSecret = v1SecretList.items.find(secret => secret.metadata!.annotations
+      && secret.metadata!.annotations['kubernetes.io/service-account.name'] === saName
```

**Comment by tolusha** [comment]
To fix the build pls apply the changes bellow
```
diff --git a/test/api/kube.test.ts b/test/api/kube.test.ts
index b073124..c45767c 100644
--- a/test/api/kube.test.ts
+++ b/test/api/kube.test.ts
@@ -100,8 +100,6 @@ describe('Kube API helper', () => {
     })
   fancy
     .nock(kubeClusterURL, api => api
-      .get('/api/v1/namespaces/default/serviceaccounts')
-      .replyWithFile(200, __dirname + '/replies/get-serviceaccounts.json', { 'Content-Type': 'application/json' })
       .get('/api/v1/namespaces/default/secrets')
       .replyWithFile(200, __dirname + '/replies/get-secrets.json', { 'Content-Type': 'application/json' })
       .get('/healthz'
```

File: src/api/kube.ts [hunk 1]
```diff
@@ -1043,51 +1043,33 @@ export class KubeHelper {
...
     const v1SecretList = res.body
-    let encodedToken
-    if (v1SecretList.items && v1SecretList.items.length > 0) {
-      for (let v1Secret of v1SecretList.items) {
-        if (v1Secret.metadata!.name === secretName && v1Secret.type === 'kubernetes.io/service-account-token') {
-          encodedToken = v1Secret.data!.token
-        }
-      }
+
+    if (!v1SecretList.items || v1SecretList.items.length <= 0) {
```
Comment (AndrienkoAleksandr) [inline src/api/kube.ts]: Is it possible `v1SecretList.items.length <= 0` ? I mean negative number value for array?

Comment (sleshchenko) [reply src/api/kube.ts]: I believe not )

**Comment by tolusha** [comment]
`test/api/replies/get-serviceaccounts.json` is missed

**Pull request merged**